### PR TITLE
Update links to n8n server

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

- Part of https://github.com/openfoodfoundation/openfoodnetwork/issues/12402

#### What should we test?

- Stage on au-staging.
- Check the database for old links. Currently, the console shows one outdated link:

```
$ cd ~/apps/openfoodnetwork/current && bundle exec rails console -e staging

ConnectedApp.pluck(:data)
=> 
[{"link"=>"https://example.net/edit"},                                                                                 
 {"link"=>"https://example.net/edit"},                                                                                 
 {"link"=>"https://example.net/edit"},                                                                                 
 {"link"=>"https://example.net/edit"},                                                                                 
 {"link"=>"https://example.net/update",                                                                                
  "destroy"=>"https://n8n.openfoodnetwork.org.uk/webhook/remove-enterprise?id=940&key=133...321b654c"},
 nil,                                                                                                                  
 nil,                                                                                                                  
 nil,                                                                                                                  
 nil]                                                                                                                  
```

That "destroy" link shouldn't have `.uk` in there any more.
#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by rioug — APPROVED** [review]
Good to see migration test :+1:

File: db/migrate/20240502035220_update_n8n_url.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# But we still have some database rows with a URL to the old server.
+class UpdateN8nUrl < ActiveRecord::Migration[7.0]
+  def up
+    execute <<~SQL.squish
+      UPDATE connected_apps
+      SET data = replace(
+        data::text,
+        'n8n.openfoodnetwork.org.uk',
+        'n8n.openfoodnetwork.org'
+      )::jsonb
```
Comment (dacook) [inline db/migrate/20240502035220_update_n8n_url.rb:13]: Nice, I learnt some new syntax today :D

File: spec/migrations/20240502035220_update_n8n_url_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require_relative '../../db/migrate/20240502035220_update_n8n_url'
+
+describe UpdateN8nUrl do
+  # We may want to move this to a support file if this syntax is useful in
+  # other places. Reference:
```
Comment (dacook) [inline spec/migrations/20240502035220_update_n8n_url_spec.rb:8]: I know I've wanted it at times. But as it's not rspec core, I avoided it and just added an expectation in a separate line below.

But I think it's unambiguous and helpful, so I'd support it 👍

**Review by dacook — APPROVED** [review]
Great work!

It's very good to have specs for data migrations.

**Comment by dacook** [comment]
✅  Staging passes:

```
irb(main):001:0> ConnectedApp.pluck(:data)
=>
[nil,
 nil,
 nil,
 nil,
 {"link"=>"https://example.net/edit"},
 {"link"=>"https://example.net/edit"},
 {"link"=>"https://example.net/edit"},
 {"link"=>"https://example.net/edit"},
 {"link"=>"https://example.net/update",
  "destroy"=>
   "https://n8n.openfoodnetwork.org/webhook/remove-enterprise?id=940&key=133bf798a9a7e112d548f1292306f311bab82885321b654c"}]
```

**Comment by dacook** [comment]
@mkllnk Is Discover Regen the only integration being used in production at the moment? I recall that OFN CA we're setting something up too; do you know if this will affect them?

**Comment by dacook** [comment]
Actually that's very easy for me to check:

```
irb(main):001:0> ConnectedApp.pluck(:data)
=> []
```

Ok, good to go!

File: spec/migrations/20240502035220_update_n8n_url_spec.rb (hunk 2 above)
Comment (rioug) [reply spec/migrations/20240502035220_update_n8n_url_spec.rb:8]: I think you could do something like : 
```ruby
#...
 }.to change { app.data["destroy"] }.to("https://n8n.openfoodnetwork.org/webhook/remove-enterprise?key=abc&id=123")
  .and change { app.data["link"] }.by(0)
```
But it's not very nice.

Comment (mkllnk) [reply spec/migrations/20240502035220_update_n8n_url_spec.rb:8]: I tried that. It works with numbers but RSpec tries to find the difference and `string - int` (minus) fails. The minus operator is also not available between two strings.

Comment (rioug) [reply spec/migrations/20240502035220_update_n8n_url_spec.rb:8]: Good to know !

**Pull request merged**